static/js/search.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from "react";
+import axios from "axios";
+import { updateUrlPlace } from "./util";
+
+let ac: google.maps.places.Autocomplete;
+
+interface PlacePropType {
+  placeName: string;
+  placeId: string;
+  deleteChip: (placeName: string, placeString: string) => void;
+}
+
+class Chip extends Component<PlacePropType, {}> {
+  render() {
+    return (
+      <span className="mdl-chip mdl-chip--deletable">
+        <span className="mdl-chip__text">{this.props.placeName}</span>
+        <button
+          className="mdl-chip__action"
+          onClick={() =>
+            this.props.deleteChip(this.props.placeName, this.props.placeId)
+          }
+        >
+          <i className="material-icons">cancel</i>
+        </button>
+      </span>
+    );
+  }
+}
+
+interface SearchBarStateType {
+  placeList: string[][];
+}
+
+class SearchBar extends Component<{}, SearchBarStateType> {
+  constructor(props) {
```

Comment:
not sure, but since there is no props, can remove this? and the super(props) below?

Reply:
It would raise error. It says on React's tutorial page "All React component classes that have a constructor should start with a super(props) call."